routers/web/repo/setting.go

```diff
@@ -349,6 +356,16 @@ func SettingsPost(ctx *context.Context) {
 			RemoteName: fmt.Sprintf("remote_mirror_%s", remoteSuffix),
 			Interval:   interval,
 		}
+		if form.PushMirrorUsePublicKey {
+			publicKey, privateKey, err := crypto.GenerateEd25519Keypair()
+			if err != nil {
+				ctx.ServerError("GenerateEd25519Keypair", err)
+				return
+			}
+			m.PrivateKey = string(privateKey)
+			m.PublicKey = string(publicKey)
+		}
+
```

Comment:
Could users delete their keys? How can a user switch an existing mirror from PublickKey to Username/Password?

Reply:
Sounds like a issue outside this PR, as you cannot change username/password either way.